lib/Lift/VariableManager.cpp

```diff
@@ -516,71 +517,79 @@ VariableManager::getByCPUStateOffsetInternal(intptr_t Offset,
   }
 }
 
-std::pair<bool, Value *> VariableManager::getOrCreate(unsigned TemporaryId,
-                                                      bool Reading) {
+std::pair<bool, Value *>
+VariableManager::getOrCreate(LibTcgArgument *Arg, bool Reading) {
   revng_assert(Instructions != nullptr);
 
-  PTCTemp *Temporary = ptc_temp_get(Instructions, TemporaryId);
-  Type *VariableType = Temporary->type == PTC_TYPE_I32 ?
+  //PTCTemp *Temporary = ptc_temp_get(Instructions, TemporaryId);
+  Type *VariableType = Arg->temp->type == LIBTCG_TYPE_I32 ?
                          AllocaBuilder.getInt32Ty() :
                          AllocaBuilder.getInt64Ty();
 
-  if (ptc_temp_is_global(Instructions, TemporaryId)) {
-    // Basically we use fixed_reg to detect "env"
-    if (Temporary->fixed_reg == 0) {
-      Value *Result = getByCPUStateOffset(EnvOffset + Temporary->mem_offset,
-                                          Temporary->name);
-      revng_assert(Result != nullptr);
-      return { false, Result };
-    } else {
-      GlobalsMap::iterator It = OtherGlobals.find(TemporaryId);
-      if (It != OtherGlobals.end()) {
+  // TODO(anjo): I guess we map TCG args to LLVM values here?
+  //             Where are constants handled?
+  switch (Arg->kind) {
+  case LIBTCG_ARG_TEMP:
+    switch (Arg->temp->kind) {
+    case LIBTCG_TEMP_EBB: {
+      // Temporary is dead at the end of the Extended Basic Block (EBB), the single entry,
```

Comment:
What's an EBB?

Reply:
The comment explains, Extended Basic Block (EBB) it's a single-entry, multiple-exit basic block where you also allow conditional branches.

It appeared in QEMU as a part of the TCG lifetime change.